sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs

```diff
@@ -137,10 +142,11 @@ public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage)
         /// </summary>
         public BinaryData Body
         {
-            get => AmqpMessage.GetBody();
+            get => body.Value;
```

Comment:
If the underlying AMQP message body is manipulated, either using this setter or directly after calling `GetRawAmqpMessage`, those changes wouldn't be reflected here.

We could offset that in the setter, but there's no `INotifyPropertyChanged` or similar on the `AmqpAnnotatedMessage` so I think we'd still have a gap.    I'm in favor of figuring out how to detect and react to changes.

I think it's potentially misleading to have the simple `Body` property doing this much active work on every get.   @JoshLove-msft:  Any thoughts?   Was this something that was discussed during arch review, by chance?

Reply:
Ah because the underlying Annotated message is mutable. Sigh... Still it seems that it is very likely that a user calls body multiple times and isn't aware of the computation and memory costs of that

Reply:
100% agree.   I think that's a valid pattern and the use of a simple `Body` property sends the message "this is lightweight and safe to call in a tight loop."   I think we need to fix this.

Reply:
I think we could detect that. by implementing the getter and setter of the AmqpAnnotatedMessage body with a custom implementation instead of the auto property it would be possible to discover those modifications

Reply:
True, but we would need to also change the API, right?

Reply:
We can optimize the 99% case without any API changes.

Reply:
Have a look at https://github.com/Azure/azure-sdk-for-net/pull/19996. This whole conversation triggered a crazy idea. We would still need to custom manage the body modifications which I think is doable without modifying the public API.

Reply:
I apologize for the fire hose of PRs

Reply:
Never apologize for being kind enough to spend your free time helping us out.  Your efforts, feedback, ideas, and discussion are very much appreciated.